src/main/java/com/google/sps/servlets/FeedServlet.java

```diff
@@ -0,0 +1,104 @@
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.json.simple.JSONObject;
+
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+
+/** Servlet that posts and gets user profiles in Datastore */
+@WebServlet("/feed")
+public class FeedServlet extends HttpServlet {
+  
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    response.setContentType("application/json");
+    JSONObject responseMap = new JSONObject();
+    ArrayList<FavoriteRecipe> recipeList = new ArrayList();
+
+    if (userService.isUserLoggedIn()) {
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      Query query = new Query("Profile")
+        .setFilter(new Query.FilterPredicate("id", Query.FilterOperator.EQUAL, userService.getCurrentUser().getUserId()));
+      PreparedQuery results = datastore.prepare(query);
+      Entity entity = results.asSingleEntity();
+
+      String email = (String) entity.getProperty("email");
+      ArrayList<String> friendList = (ArrayList<String>) entity.getProperty("friendList");
+      friendList = nullToArrayList(friendList);
+
+      for (String friend: friendList) {
+        Query emailQuery = new Query("Profile")
+          .setFilter(new Query.FilterPredicate("email", Query.FilterOperator.EQUAL, friend));
+        PreparedQuery emailResults = datastore.prepare(emailQuery);
+        Entity emailEntity = emailResults.asSingleEntity();
+        String userId = (String) emailEntity.getProperty("id");
+
+        Query userQuery = new Query("FavoriteRecipe")
+          .setFilter(new Query.FilterPredicate("userId", Query.FilterOperator.EQUAL, userId));
+        PreparedQuery userResults = datastore.prepare(userQuery);
+        for (Entity userEntity : userResults.asIterable()) {
+          long entityFavoriteId = userEntity.getKey().getId();
+          System.out.println(entityFavoriteId);
+          String entityUserId = (String) friend;
+          long entityRecipeId = (long) userEntity.getProperty("recipeId");
+          String entityDate = (String) userEntity.getProperty("dateFavorited");
+
+          FavoriteRecipe testing = new FavoriteRecipe (entityFavoriteId, friend, entityRecipeId, entityDate);
+          recipeList.add(testing);
+        }
+      }
+      responseMap.put("recipeList", recipeList);
+      responseMap.put("email", email);
+    } else {
+      String errorMessage = "User needs to log in.";
+      responseMap.put("error", errorMessage);
+    }
```

Comment:
nit: you may want to move this block at the top to improve readability, i.e. if (user not logged in ) {} else {}